cranelift/codegen/src/machinst/abi.rs

```diff
@@ -608,26 +608,26 @@ pub trait ABIMachineSpec {
 pub struct Sig(u32);
 cranelift_entity::entity_impl!(Sig);
 
+impl Sig {
+    fn prev(self) -> Option<Sig> {
+        self.0.checked_sub(1).map(Sig)
+    }
+}
+
 /// ABI information shared between body (callee) and caller.
 #[derive(Clone, Debug)]
 pub struct SigData {
-    /// Argument location starting offset (regs or stack slots). Stack offsets are relative to
+    /// Argument location ending offset (regs or stack slots). Stack offsets are relative to
```

Comment:
Can we add a note here on the indexing scheme? Something like "Arguments for a given `sig` are in the args array at indicies `args_end[sig - 1]..args_end[sig]`, or `0..args_end[sig]` if `sig` is `0`.

(And the `args_end` value is an exclusive endpoint, i.e. not included, so `..` above is correct, right?)

Reply:
Or, reading more below, I guess it's a little more complex because args and rets are contiguous for a given `Sig`, but we should give the appropriate ranges here and describe the scheme regardless.

Reply:
@cfallin added more comments, let me know if this looks good?

Reply:
Looks great, thanks!